flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -177,6 +187,11 @@ public void notifyCheckpointComplete(long completedCheckpointId) {
 				materializedSstFiles.keySet().removeIf(checkpointId -> checkpointId < completedCheckpointId);
 				lastCompletedCheckpointId = completedCheckpointId;
 			}
+
+			// We need to close the file for current completed checkpoint.
+			if (checkpointStreamFactory instanceof FsSegmentCheckpointStreamFactory) {
```

Comment:
I think maybe we also need to close the file for current aborted checkpoint too.